sql/src/main/java/io/crate/lucene/LuceneQueryBuilder.java

```diff
@@ -1126,6 +1127,57 @@ public Query apply(Function parent, Function inner, Context context) {
             }
         }
 
+        static class SubscriptQuery implements InnerFunctionToQuery {
+
+            private static final ImmutableMap<String, BooleanClause.Occur> FUNCTION_TO_CLAUSE_OCCURRENCE =
```

Comment:
Without reading the code below it's not visible why `Gt` and `Lt` have `MUST_NOT` but all others have `MUST`. Maybe we can clarify that a bit. Maybe renaming to something like `PRE_FILTER_OCCURRENCE_BY_OP` or `NARROWING_CLAUSE_BY_OP`  would help to indicate that these are for the additional "pre-filter" clauses.